README.md

```diff
@@ -22,36 +36,248 @@ component written in native code. Using that component from DNX
 currently [requires a workaround][grpc-workaround]; on Windows please run `fix_dnx_grpc.bat`
 before running any gRPC code.
 
-Documentation
--------------
+## Specifying a Project ID
+
+Most `gcloud-dotnet` libraries require a project ID. If you don't remember yours (or haven't created a project yet), navigate to the [Google Developers Console][google-developers-console] to view your project ID (or create a new project and get the ID then). Once done, there are multiple ways to specify the project ID for the client libraries to retrieve the value:
+
+1. When using `gcloud-dotnet` libraries from within Compute/App Engine, there's no need to specify a project ID.  It is automatically inferred from the production environment.
+2. When using `gcloud-dotnet` libraries elsewhere, you can do one of the following:
+    * Define the environment variable GCLOUD_PROJECT to be your desired project ID. For example:
+
+      ```
+      set GCLOUD_PROJECT=PROJECT_ID
+      ```
+    * If running locally for development/testing, set the project ID using the [Google Cloud SDK][google-cloud-sdk].  Download the SDK if you haven't already, and set the project ID by running the following in the command line:
+
+      ```
+      gcloud config set project PROJECT_ID
+      ```
+
+## Authentication
+Every API call needs to be authenticated. In other to successfully make a call, first ensure that the necessary Google Cloud APIs are enabled for your project and that you've downladed the right set of keys (if it applies to you) as explained in the [authentication document][cloud-common-authentication].
+
+Next, choose a method for authenticating API requests from within your project:
+
+1. When using `gcloud-dotnet` libraries from within Compute/App Engine, no additional authentication steps are necessary.
+2. When using `gcloud-dotnet` libraries elsewhere, you can do one of the following:
+    * Define the environment variable GOOGLE_APPLICATION_CREDENTIALS to be the location of the key.  For example:
+
+      ```
+      set GOOGLE_APPLICATION_CREDENTIALS=/path/to/my/key.json
+      ``` 
+    * If running locally for development/testing, you can authenticate using the [Google Cloud SDK][google-cloud-sdk].  Download the SDK if you haven't already, then login by running the following in the command line:
+
+      ```
+      gcloud auth login
+      ```
+
+## Google BigQuery
+
+- [API Reference][cloud-bigquery-ref]
+- [API Overview][cloud-bigquery-docs]
+
+The BigQuery client library is in pre-release state and is more likely to have significant surface changes over time. Still, if you would like to experiment with it, we would welcome your feedback.
+
+#### Installation
+The package is available on Google's [public MyGet feed][google-myget-feed], where you'll also find the right Package Source to add in Visual Studio. Once done, the package can be installed in the following manner:
+
+```sh
+PM> Install-Package Google.Bigquery.V2 -Prerelease
+```
+
+#### Example
```

Comment:
So I agree with you (and I had that conversation yesterday with Jonathan) that having the examples written in two places does certainly feel like duplication - and more work to keep in sync. However, showing the snippets in two places feels great: meaning, having the snippet there in the ReadMe does help and gives confidence that this repo is alive and usable. Having a link to see the snippet somewhere else is just more text (and less code).

So is there a way to have the code showing in both the ReadMe and in the Client API Ref source from the same place? That would be the ideal: one code location and multiple views.